src/main/java/training/chessington/model/Board.java

```diff
@@ -50,4 +52,56 @@ public void placePiece(Coordinates coords, Piece piece) {
         board[coords.getRow()][coords.getCol()] = piece;
     }
 
+    public boolean moveIsBlocked(Move move) {
+        
+        int fromRow = move.getFrom().getRow();
+        int fromCol = move.getFrom().getCol();
+        int toRow = move.getTo().getRow();
+        int toCol = move.getTo().getCol();
+        
+        boolean isHorizontal = fromRow == toRow;
+        boolean isVertical = fromCol == toCol;
+        boolean isDiagonal = (fromRow - toRow) == (fromCol - toCol);

```

Comment:
I expect something about this line will pop up when you come to writing tests for bishops...